src/components/calcite-chip/calcite-chip.tsx

```diff
@@ -79,28 +79,70 @@ export class CalciteChip {
         : "l";
 
     const iconEl = (
-      <calcite-icon
-        class="calcite-chip--icon"
-        icon={this.icon}
-        scale={iconScale}
-      />
+      <calcite-icon class={this.iconClass} icon={this.icon} scale={iconScale} />
     );
-
     return (
       <Host dir={dir}>
         {this.icon ? iconEl : null}
         <slot name="chip-image"></slot>
-        <span>
+        <span class={this.spanClass}>
           <slot />
         </span>
         <button
           onClick={this.closeClickHandler}
-          class={CSS.close}
+          class={this.buttonClass}
           title={TEXT.close}
         >
           <calcite-icon scale={iconScale} icon="x" />
         </button>
       </Host>
     );
   }
+
+  private get iconClass() {
```

Comment:
I thought putting the class functions at the end of the file would be a good convention to follow.  They're pretty long, so this could help keep the style stuff a little less intrusive and out of the way when you're not wanting to look at it.